pyiron_contrib/atomistics/atomicrex/structure_list.py

```diff
@@ -8,64 +8,316 @@
+#        self.cell = cell
+#        self.symbols = symbols
+#        self.positions = positions
 
-class StructureList(object):
+
+class FlattenedStructureContainer:
     """
-    Possible container class to inherit from in other potential fitting interfaces.
-    Probably obsolete if StructureContainer or something similar gets developed.
+    Class that can write and read lots of structures from and to hdf quickly.
+    This is done by storing positions, cells, etc. into large arrays instead of writing every structure into a new group.
+    Structures are stored together with an identifier that should be unique.
+    The class has to be initialized with the number of structures and the total number of atoms in all structures.
+    This allows to preallocate the necessary arrays and therefore prevents reallocations.
+
+    TODO: Alternatively preallocate arrays with large default values (f.e. 1e6 atoms) then use resize to shrink them?
     """    
-    def __init__(self):
-        self._structure_lst = []
-
-    def add_structure(self, structure, **kw_properties):
-        if isinstance(structure, ASEAtoms):
-            i = structure.info
-            structure = ase_to_pyiron(structure)
-            structure.info = i
-        elif isinstance(structure, Atoms):
-            pass
-        else:
-            raise ValueError("Structures have to be supplied as pyiron or ase atoms")
+    def __init__(self, num_structures=0, num_atoms=0):
+        self.num_structures = num_structures
+        self.num_atoms = num_atoms
+        # store the starting index for properties with unknown length
+        self.current_atom_index = 0
+        # store the index for properties of known size, stored at the same index as the structure
+        self.current_structure_index = 0
+        # Also store indices of structure recently added
+        self.prev_structure_index = 0
+        self.prev_atom_index = 0
+        
+        self._init_arrays()
+
+
+    def _init_arrays(self):
+        self.symbols = np.full(self.num_atoms, "XX", dtype=np.dtype("U2"))# 2 character unicode array for chemical symbols
+        self.positions = np.empty((self.num_atoms, 3))
+        self.cells = np.empty((self.num_structures, 3, 3))
+        self.start_indices = np.empty(self.num_structures, dtype=np.int32)
+        self.len_current_struct = np.empty(self.num_structures, dtype=np.int32)
+        self.identifiers = np.empty(self.num_atoms, dtype=np.dtype("U20"))
+
+
+    def add_structure(self, structure, identifier):
+        # len of structure to index into the initialized arrays
+        n = len(structure)
+        i = self.current_atom_index + n
+
+        self.len_current_struct[self.current_structure_index] = n
+        self.symbols[self.current_atom_index:i] = np.array(structure.symbols)
+        self.positions[self.current_atom_index:i] = structure.positions
+        self.cells[self.current_atom_index:i] = structure.cell.array
```

Comment:
```suggestion
        self.cells[self.current_structure_index] = structure.cell.array
```

Reply:
how did this even work and not throw an index error